### DISCUSSION TITLE
How to use ViewAction in my EditRessource ?

### DISCUSSION THREAD

**Discussion by ludo-git** [filamentphp/filament#5753] [Help] [answered]
When i call in my Ressource :

```
public static function table(Table $table): Table
{
    return $table
        ->actions([
            ViewAction::make()]);
}
```
I do have a popup showing me the fields listed on 

`public static function form(Form $form): Form`

How can i achieve the same in a button in my custom EditRessource ?

I tried here : 

```
protected function getActions(): array
{
  return [
      ViewAction::make('View')
          ->label('View')
  ];
}
```
But nothing is happening.

**Comment by Congminh7799** [comment]
U can use:
`
                 ->action(function () {
                            redirect(static::getResource()::getUrl('view', ['record' => $this->record]));
                 })
`

**Reply by danharrin** [reply]
Pass a url() instead of using a redirect

**Reply by Congminh7799** [reply]
sorry, because i use ->requiresConfirmation(), if i use url(), ->requiresConfirmation inactive.

**Reply by danharrin** [reply]
Ah ok

**Comment by ludo-git** [comment, marked as answer]
I managed to do what i wanted with the following code :

```
Action::make('View')
    ->mountUsing(function (ComponentContainer $form): void {
        $data = $this->record->attributesToArray();
        $form->fill($data);
    })
    ->form($this->getFormSchema())
    ->modalHeading(__('View'))
    ->disableForm()
    ->action(static function (): void {})
    ->modalActions([])
    ->icon('heroicon-o-eye'),
```
